compiler/lexer.cpp

```diff
@@ -780,8 +812,9 @@ void TokenLexerHeredocString::init() {
   h->add_rule("\\[0-7]", &vk::singleton<TokenLexerOctChar>::get());
   h->add_rule("\\x[0-9A-Fa-f]", &vk::singleton<TokenLexerHexChar>::get());
 
-  h->add_rule("$[A-Za-z{]", &vk::singleton<TokenLexerStringSimpleExpr>::get());
+  h->add_rule("$[A-Za-z]", &vk::singleton<TokenLexerStringSimpleExpr>::get());
```

Comment:
it seems like a subtle bug here is hidden, maybe we unify these Lexers?
![image](https://user-images.githubusercontent.com/5980005/99806357-fdc63180-2b4e-11eb-9604-abfdde5c0d3c.png)

Reply:
Fixed. Relevant tests are added.